tests/tests/test-precompile-erc20.ts

```diff
@@ -0,0 +1,258 @@
+    data: `0x${SELECTORS.approve}${spenderData}${amount}`,
+  });
+
+  const block = await context.createBlock({
+    transactions: [tx],
+  });
+
+  const receipt = await context.web3.eth.getTransactionReceipt(block.txResults[0].result);
+  expect(receipt.status).to.equal(true);
+  expect(receipt.logs.length).to.eq(1);
+  expect(receipt.logs[0].address).to.eq(ADDRESS_ERC20);
+  expect(receipt.logs[0].data).to.eq(`0x${amount}`);
+  expect(receipt.logs[0].topics.length).to.eq(3);
+  expect(receipt.logs[0].topics[0]).to.eq(SELECTORS.logApprove);
+  expect(receipt.logs[0].topics[1]).to.eq(`0x${fromData}`);
+  expect(receipt.logs[0].topics[2]).to.eq(`0x${spenderData}`);
+}
+
+async function checkAllowance(context, owner, spender, amount) {
+  const ownerData = owner.slice(2).padStart(64, "0");
+  const spenderData = spender.slice(2).padStart(64, "0");
+
+  const request = await customWeb3Request(context.web3, "eth_call", [
+    {
+      from: GENESIS_ACCOUNT,
+      value: "0x0",
+      gas: "0x10000",
+      gasPrice: GAS_PRICE,
+      to: ADDRESS_ERC20,
+      data: `0x${SELECTORS.allowance}${ownerData}${spenderData}`,
+    },
+  ]);
+
+  expect(request.result).equals(`0x${amount.padStart(64, "0")}`);
+}
+
+describeDevMoonbeam("Precompiles - ERC20 Native", (context) => {
+  it("allows to call getBalance", async function () {
+    const address = ALITH.slice(2).padStart(64, "0");
+
+    const tx_call = await customWeb3Request(context.web3, "eth_call", [
+      {
+        from: GENESIS_ACCOUNT,
+        value: "0x0",
+        gas: "0x10000",
+        gasPrice: GAS_PRICE,
+        to: ADDRESS_ERC20,
+        data: `0x${SELECTORS.balanceOf}${address}`,
+      },
+    ]);
+
+    let amount = await getBalance(context, 0, ALITH);
+    amount = "0x" + amount.toHex().slice(2).padStart(64, "0");
+    expect(tx_call.result).equals(amount);
+  });
+
+  it("allows to call totalSupply", async function () {
+    const tx_call = await customWeb3Request(context.web3, "eth_call", [
+      {
+        from: GENESIS_ACCOUNT,
+        value: "0x0",
+        gas: "0x10000",
+        gasPrice: GAS_PRICE,
+        to: ADDRESS_ERC20,
+        data: `0x${SELECTORS.totalSupply}`,
+      },
+    ]);
+
+    const amount = await context.polkadotApi.query.balances.totalIssuance();
+    const amount_hex = "0x" + amount.toHex().slice(2).padStart(64, "0");
+
+    console.log({ tx_call });
+
+    expect(tx_call.result).equals(amount_hex);
+  });
+});
+
+describeDevMoonbeam("Precompiles - ERC20 Native", (context) => {
+  it("allows to approve transfers, and allowance matches", async function () {
+    const amount = `1000000000000`.padStart(64, "0");
+
+    await sendApprove(context, ALITH, ALITH_PRIV_KEY, BALTATHAR, amount);
+
+    await checkAllowance(context, ALITH, BALTATHAR, amount);
+  });
+});
+
+describeDevMoonbeam("Precompiles - ERC20 Native", (context) => {
+  it("allows to call transfer", async function () {
+    const amount = `400000000000`.padStart(64, "0");
```

Comment:
WOuld be nice to use constant and convert to hex

Reply:
Here it's just an arbitrary value to test transfers, and I wanted to avoid conversions issues due to large numbers.